dev/ready-probe-labeler/main.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package main
+
+import (
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/pflag"
+	"golang.org/x/net/context"
+	"golang.org/x/xerrors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+)
+
+var (
+	// ServiceName is the name we use for tracing/logging
+	ServiceName = "ready-probe-labeler"
+	// Version of this service - set during build
+	Version = ""
+)
+
+var opts struct {
+	JSONLog    bool
+	Verbose    bool
+	Shutdown   bool
+	Kubeconfig string
+	Label      string
+	ProbeURL   string
+	Timeout    time.Duration
+}
+
+func main() {
+	pflag.BoolVarP(&opts.JSONLog, "json-log", "j", true, "produce JSON log output on verbose level")
+	pflag.BoolVarP(&opts.Shutdown, "shutdown", "s", false, "Removes the node label previous to a pod stop")
+	pflag.StringVarP(&opts.Label, "label", "l", "", "Gitpod component Label")
+	pflag.BoolVarP(&opts.Verbose, "verbose", "v", false, "Enable verbose JSON logging")
+	pflag.StringVarP(&opts.Kubeconfig, "kubeconfig", "k", "", "Kubeconfig file path")
+	pflag.StringVarP(&opts.ProbeURL, "probe-url", "p", "http://localhost:9501/ready", "URL to test the readiness of a service")
+	pflag.DurationVarP(&opts.Timeout, "timeout", "t", 60*time.Second, "W")
+	pflag.Parse()
+
+	log.Init(ServiceName, Version, opts.JSONLog, opts.Verbose)
+
+	log.Info("Starting node labeler...")
+
+	if opts.Label == "" {
+		log.Fatalf("Please set the flag --label")
+	}
+
+	nodeName := os.Getenv("NODENAME")
+	if nodeName == "" {
+		log.Fatalf("Environment variable NODENAME is not defined")
+	}
+
+	client, err := getKubeClient(opts.Kubeconfig)
+	if err != nil {
+		log.Fatalf("Unexpected error: %v", err)
+	}
+
+	if opts.Shutdown {
+		err = updateLabel(opts.Label, false, nodeName, client)
+		if err != nil {
+			log.Fatalf("Unexpected error removing node label: %v", err)
+		}
+
+		return
+	}
+
+	time.Sleep(1 * time.Second)
```

Comment:
what is the reason for 1s sleep here?

Reply:
No particular reason besides not starting right away